src/main/java/com/dataartisans/flink/dataflow/translation/wrappers/streaming/state/StateType.java

```diff
@@ -0,0 +1,56 @@
+package com.dataartisans.flink.dataflow.translation.wrappers.streaming.state;
+
+import java.io.IOException;
+
+public enum StateType {
+
+	VALUE,
+
+	WATERMARK,
+
+	LIST,
+
+	ACCUMULATOR;
+
+	public static void serialize(StateType type, StateCheckpointWriter output) throws IOException {
+		if (output == null) {
+			throw new IllegalArgumentException("Cannot write to a null output.");
+		}
+
+		switch (type) {
+			case VALUE:
+				output.writeByte((byte) 0);
```

Comment:
You could assign numbers to the possible values of the Enum in the constructor. That way you don't loose track in the serialize/deserialize methods.